# <p style="color:tomato">Summarizing Data</p>

data_sorce: U.S. Cencus Bureau survey <br>
data: everything from housing affordability to industry employment rates <br>
cleaned_data_from : FiveThirtyEight <br>
[https://github.com/fivethirtyeight/data/tree/master/college-majors](https://github.com/fivethirtyeight/data/tree/master/college-majors) <br>
cleaned_data : 2010-2012/ all-ages.csv/ recent-grads.csv

<p> <b>Read</b> Data file</p>
<p> <b>Assign</b> the value</p>

In [1]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [2]:
import pandas as pd
import numpy as np

In [3]:
all_ages = pd.read_csv('all-ages.csv')
# all_ages

In [4]:
recent_grads = pd.read_csv('recent-grads.csv')
# recent_grads

<p> <b>Count</b> the number of people in each Major_category for both data sets.</p>
<p style="color:red">series.unique()</p>
: return the unique values in a colomn (major_category)

In [5]:
print(recent_grads['Major_category']) # 정렬이 되지 않은채 모든 row가 나온다. 
print(recent_grads['Major_category'].unique()) # 컬럼 Major_category에 있는 unique한 값만 나온다. 

0                              Engineering
1                              Engineering
2                              Engineering
3                              Engineering
4                              Engineering
5                              Engineering
6                                 Business
7                        Physical Sciences
8                              Engineering
9                              Engineering
10                             Engineering
11                             Engineering
12                             Engineering
13                             Engineering
14                             Engineering
15                             Engineering
16                             Engineering
17                             Engineering
18                             Engineering
19                     Law & Public Policy
20                 Computers & Mathematics
21         Agriculture & Natural Resources
22                             Engineering
23         

<p> <b>Make</b> a function for each unique value</p>
<p> return all of the rows where the column equals tht unique value. </p>


<p style="color:red">Dictionary</p>

: dictionary에 대해서 알아보기 <br>

### <p style="color:red">How to convert Data Frame data to dictionary data</p>

In [6]:
print(all_ages['Major_category'].unique())

['Agriculture & Natural Resources' 'Biology & Life Science' 'Engineering'
 'Humanities & Liberal Arts' 'Communications & Journalism'
 'Computers & Mathematics' 'Industrial Arts & Consumer Services'
 'Education' 'Law & Public Policy' 'Interdisciplinary' 'Health'
 'Social Science' 'Physical Sciences' 'Psychology & Social Work' 'Arts'
 'Business']


In [7]:
print(recent_grads['Major_category'].unique())

['Engineering' 'Business' 'Physical Sciences' 'Law & Public Policy'
 'Computers & Mathematics' 'Agriculture & Natural Resources'
 'Industrial Arts & Consumer Services' 'Arts' 'Health' 'Social Science'
 'Biology & Life Science' 'Education' 'Humanities & Liberal Arts'
 'Psychology & Social Work' 'Communications & Journalism'
 'Interdisciplinary']


In [8]:
recent_grads.head()

Rank  Major_code                                      Major    Total  \
0     1        2419                      PETROLEUM ENGINEERING   2339.0   
1     2        2416             MINING AND MINERAL ENGINEERING    756.0   
2     3        2415                  METALLURGICAL ENGINEERING    856.0   
3     4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   1258.0   
4     5        2405                       CHEMICAL ENGINEERING  32260.0   

       Men    Women Major_category  ShareWomen  Sample_size  Employed  \
0   2057.0    282.0    Engineering    0.120564           36      1976   
1    679.0     77.0    Engineering    0.101852            7       640   
2    725.0    131.0    Engineering    0.153037            3       648   
3   1123.0    135.0    Engineering    0.107313           16       758   
4  21239.0  11021.0    Engineering    0.341631          289     25694   

       ...        Part_time  Full_time_year_round  Unemployed  \
0      ...              270                  1207          37   
1      ...              170                   388          85   
2      ...              133                   340          16   
3      ...              150                   692          40   
4      ...             5180                 16697        1672   

   Unemployment_rate  Median  P25th   P75th  College_jobs  Non_college_jobs  \
0           0.018381  110000  95000  125000          1534               364   
1           0.117241   75000  55000   90000           350               257   
2           0.024096   73000  50000  105000           456               176   
3           0.050125   70000  43000   80000           529               102   
4           0.061098   65000  50000   75000         18314              4440   

   Low_wage_jobs  
0            193  
1             50  
2              0  
3              0  
4            972  

[5 rows x 21 columns]

## summarizing Major Categories
Chellenge: summarizing Data <br>
summarizing Major categories

In [10]:
# aa_cat_counts = dict()
# 왜 안될까?
aa_cat_counts = {}
# rg_cat_counts = dict()
rg_cat_counts = {}

In [11]:
aa_cats = all_ages['Major_category'].unique()
for c in aa_cats:
    major_df = all_ages[all_ages['Major_category'] == c]
    total = major_df['Total'].sum()
    aa_cat_counts[c] = total
print(aa_cat_counts)

{'Agriculture & Natural Resources': 632437, 'Biology & Life Science': 1338186, 'Engineering': 3576013, 'Humanities & Liberal Arts': 3738335, 'Communications & Journalism': 1803822, 'Computers & Mathematics': 1781378, 'Industrial Arts & Consumer Services': 1033798, 'Education': 4700118, 'Law & Public Policy': 902926, 'Interdisciplinary': 45199, 'Health': 2950859, 'Social Science': 2654125, 'Physical Sciences': 1025318, 'Psychology & Social Work': 1987278, 'Arts': 1805865, 'Business': 9858741}


In [12]:
rg_cats = recent_grads['Major_category'].unique()
for c in rg_cats:
    mojor_df = recent_grads[recent_grads['Major_category'] == c]
    total = major_df['Total'].sum()
    rg_cat_counts[c] = total
print(rg_cat_counts)

{'Engineering': 9858741, 'Business': 9858741, 'Physical Sciences': 9858741, 'Law & Public Policy': 9858741, 'Computers & Mathematics': 9858741, 'Agriculture & Natural Resources': 9858741, 'Industrial Arts & Consumer Services': 9858741, 'Arts': 9858741, 'Health': 9858741, 'Social Science': 9858741, 'Biology & Life Science': 9858741, 'Education': 9858741, 'Humanities & Liberal Arts': 9858741, 'Psychology & Social Work': 9858741, 'Communications & Journalism': 9858741, 'Interdisciplinary': 9858741}


In [13]:
# print(all_ages['Major_category'].unique())
cats = all_ages['Major_category'].unique()
print(cats)
counts_dictionary = {}
for c in cats:
    major_df = all_ages[all_ages['Major_category'] == c]
    total = major_df["Total"].sum()
    counts_dictionary[c] = total
    
print(counts_dictionary)

['Agriculture & Natural Resources' 'Biology & Life Science' 'Engineering'
 'Humanities & Liberal Arts' 'Communications & Journalism'
 'Computers & Mathematics' 'Industrial Arts & Consumer Services'
 'Education' 'Law & Public Policy' 'Interdisciplinary' 'Health'
 'Social Science' 'Physical Sciences' 'Psychology & Social Work' 'Arts'
 'Business']
{'Agriculture & Natural Resources': 632437, 'Biology & Life Science': 1338186, 'Engineering': 3576013, 'Humanities & Liberal Arts': 3738335, 'Communications & Journalism': 1803822, 'Computers & Mathematics': 1781378, 'Industrial Arts & Consumer Services': 1033798, 'Education': 4700118, 'Law & Public Policy': 902926, 'Interdisciplinary': 45199, 'Health': 2950859, 'Social Science': 2654125, 'Physical Sciences': 1025318, 'Psychology & Social Work': 1987278, 'Arts': 1805865, 'Business': 9858741}


In [14]:
def calculate_major_cat_totals(df):
    cats = df['Major_category'].unique()
    counts = dict()
    for c in cats:
        major_df = df[df["Major_category"] == c]
        total = major_df["Total"].sum()
        return counts

In [15]:
list(recent_grads['Major_category'].unique())

['Engineering',
 'Business',
 'Physical Sciences',
 'Law & Public Policy',
 'Computers & Mathematics',
 'Agriculture & Natural Resources',
 'Industrial Arts & Consumer Services',
 'Arts',
 'Health',
 'Social Science',
 'Biology & Life Science',
 'Education',
 'Humanities & Liberal Arts',
 'Psychology & Social Work',
 'Communications & Journalism',
 'Interdisciplinary']

In [16]:
print(recent_grads['Major_category'].unique().tolist())

['Engineering', 'Business', 'Physical Sciences', 'Law & Public Policy', 'Computers & Mathematics', 'Agriculture & Natural Resources', 'Industrial Arts & Consumer Services', 'Arts', 'Health', 'Social Science', 'Biology & Life Science', 'Education', 'Humanities & Liberal Arts', 'Psychology & Social Work', 'Communications & Journalism', 'Interdisciplinary']


In [17]:
aa_cat_counts = dict()
print(aa_cat_counts)

{}


In [18]:
dict = {'Name': 'Zara', 'Age': 7, 'Class': 'First'}

In [19]:
print(dict)

{'Name': 'Zara', 'Age': 7, 'Class': 'First'}


In [20]:
dict.items()
dict.keys()

dict_items([('Name', 'Zara'), ('Age', 7), ('Class', 'First')])

dict_keys(['Name', 'Age', 'Class'])

In [21]:
recent_grads['Major_category'].unique()

array(['Engineering', 'Business', 'Physical Sciences',
       'Law & Public Policy', 'Computers & Mathematics',
       'Agriculture & Natural Resources',
       'Industrial Arts & Consumer Services', 'Arts', 'Health',
       'Social Science', 'Biology & Life Science', 'Education',
       'Humanities & Liberal Arts', 'Psychology & Social Work',
       'Communications & Journalism', 'Interdisciplinary'], dtype=object)

In [24]:
# Unique values in Major_category column.
print(all_ages['Major_category'].unique())

aa_cats = all_ages['Major_category'].unique()
aa_cat_counts = {}
for c in aa_cats:
    major_df = all_ages[all_ages['Major_category'] == c]
    total = major_df["Total"].sum()
    aa_cat_counts[c] = total
print(aa_cat_counts)

rg_cats = recent_grads['Major_category'].unique()

rg_cat_counts = {}

for c in rg_cats:
    major_df = recent_grads[recent_grads['Major_category'] == c]
    total = major_df['Total'].sum()
    rg_cat_counts[c] = total
print(rg_cat_counts)
    

['Agriculture & Natural Resources' 'Biology & Life Science' 'Engineering'
 'Humanities & Liberal Arts' 'Communications & Journalism'
 'Computers & Mathematics' 'Industrial Arts & Consumer Services'
 'Education' 'Law & Public Policy' 'Interdisciplinary' 'Health'
 'Social Science' 'Physical Sciences' 'Psychology & Social Work' 'Arts'
 'Business']
{'Agriculture & Natural Resources': 632437, 'Biology & Life Science': 1338186, 'Engineering': 3576013, 'Humanities & Liberal Arts': 3738335, 'Communications & Journalism': 1803822, 'Computers & Mathematics': 1781378, 'Industrial Arts & Consumer Services': 1033798, 'Education': 4700118, 'Law & Public Policy': 902926, 'Interdisciplinary': 45199, 'Health': 2950859, 'Social Science': 2654125, 'Physical Sciences': 1025318, 'Psychology & Social Work': 1987278, 'Arts': 1805865, 'Business': 9858741}
{'Engineering': 537583.0, 'Business': 1302376.0, 'Physical Sciences': 185479.0, 'Law & Public Policy': 179107.0, 'Computers & Mathematics': 299008.0, 'Agric

* 언론매체 저임금, 비숙련 직업을 구하는 대학 졸업자의 수에 대해 이야기
* 데이터 담당자는 광범위한 주장에 회의적이어야 한다. 

In [25]:
all_ages.head()

Major_code                                  Major  \
0        1100                    GENERAL AGRICULTURE   
1        1101  AGRICULTURE PRODUCTION AND MANAGEMENT   
2        1102                 AGRICULTURAL ECONOMICS   
3        1103                        ANIMAL SCIENCES   
4        1104                           FOOD SCIENCE   

                    Major_category   Total  Employed  \
0  Agriculture & Natural Resources  128148     90245   
1  Agriculture & Natural Resources   95326     76865   
2  Agriculture & Natural Resources   33955     26321   
3  Agriculture & Natural Resources  103549     81177   
4  Agriculture & Natural Resources   24280     17281   

   Employed_full_time_year_round  Unemployed  Unemployment_rate  Median  \
0                          74078        2423           0.026147   50000   
1                          64240        2266           0.028636   54000   
2                          22810         821           0.030248   63000   
3                          64937        3619           0.042679   46000   
4                          12722         894           0.049188   62000   

   P25th    P75th  
0  34000  80000.0  
1  36000  80000.0  
2  40000  98000.0  
3  30000  72000.0  
4  38500  90000.0

In [26]:
recent_grads.head()

Rank  Major_code                                      Major    Total  \
0     1        2419                      PETROLEUM ENGINEERING   2339.0   
1     2        2416             MINING AND MINERAL ENGINEERING    756.0   
2     3        2415                  METALLURGICAL ENGINEERING    856.0   
3     4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   1258.0   
4     5        2405                       CHEMICAL ENGINEERING  32260.0   

       Men    Women Major_category  ShareWomen  Sample_size  Employed  \
0   2057.0    282.0    Engineering    0.120564           36      1976   
1    679.0     77.0    Engineering    0.101852            7       640   
2    725.0    131.0    Engineering    0.153037            3       648   
3   1123.0    135.0    Engineering    0.107313           16       758   
4  21239.0  11021.0    Engineering    0.341631          289     25694   

       ...        Part_time  Full_time_year_round  Unemployed  \
0      ...              270                  1207          37   
1      ...              170                   388          85   
2      ...              133                   340          16   
3      ...              150                   692          40   
4      ...             5180                 16697        1672   

   Unemployment_rate  Median  P25th   P75th  College_jobs  Non_college_jobs  \
0           0.018381  110000  95000  125000          1534               364   
1           0.117241   75000  55000   90000           350               257   
2           0.024096   73000  50000  105000           456               176   
3           0.050125   70000  43000   80000           529               102   
4           0.061098   65000  50000   75000         18314              4440   

   Low_wage_jobs  
0            193  
1             50  
2              0  
3              0  
4            972  

[5 rows x 21 columns]

In [27]:
recent_grads['Low_wage_jobs'].head()

0    193
1     50
2      0
3      0
4    972
Name: Low_wage_jobs, dtype: int64

In [29]:
recent_grads['Total'].head()

0     2339.0
1      756.0
2      856.0
3     1258.0
4    32260.0
Name: Total, dtype: float64

Low_wage_jobs. and Toltal. columns to calculate the proportion of recent college graduates that worked low wage jobs. 
* proportion : 비율

#### Series.sum()

In [31]:
low_wage_percent = 0.0
low_wage_jobs_sum = recent_grads['Low_wage_jobs'].sum()
print(low_wage_jobs_sum)

667610


In [32]:
recent_grads_sum = recent_grads['Total'].sum()
print(recent_grads_sum)

6771654.0


In [33]:
low_wage_proportion = low_wage_jobs_sum / recent_grads_sum
print(low_wage_proportion)

0.09858891195563152


0.09858891195563152 means!!
* only about 9.85%, equivalent to a proportion of 0.0985, of graduates took on a low wage job after finishing college.

* all_ages and recent_grads have 173rows
* 두 데이터 세트 사이의 비교를 수행
* 최근 대학 졸업자의 통계가 전체 인구 통계와 어떻게 비교되는지 확인
* Let's calculate the number of majors where recent graduates did better than the overall population.

Boolean filtering 

In [46]:
major = recent_grads['Major'].unique()

In [45]:
rg_lower_count = 0
for all_age in range(len(all_ages['Major'])):
    for recent in range(len(recent_grads['Major'])):
#         print(recent_grads['Major'][recent])
        if all_ages['Major'][all_age] == recent_grads['Major'][recent]:
#             print(all_ages['Unemployment_rate'])
#             print(recent_grads['Unemployment_rate'])
            if all_ages['Unemployment_rate'][all_age] > recent_grads['Unemployment_rate'][recent]:
                rg_lower_count += 1
print(rg_lower_count)

44


In [58]:
for m in major:
    recent_grads_row = recent_grads[recent_grads['Major'] == m]
    all_ages_row = all_ages[all_ages['Major'] == m]
#     print(recent_grads_row)
    rg_unemp_rate = recent_grads_row.iloc[0]['Unemployment_rate']
    aa_unemp_rate = all_ages_row.iloc[0]['Unemployment_rate']
#     print(recent_grads_row.iloc[0])
#     print(type(recent_grads_row.iloc[0]))
    if rg_unemp_rate < aa_unemp_rate:
        rg_lower_count += 1
print(rg_lower_count)

<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.Series'>
<class 'pandas.core.series.S